docs/quick-release-notes/main.go

```diff
@@ -0,0 +1,314 @@
+	var versionTitles []string
+	var versionNumbers []string
+	for scanner.Scan() {
+		changeLogLength++
+		changeLogText = append(changeLogText, scanner.Text())
+		if re.MatchString(scanner.Text()) {
+			hashVersionAndDate := re.FindStringSubmatch(scanner.Text())
+			idxVersionNumberAndDate := re.SubexpIndex("versionNumberAndDate")
+			idxVersionNumber := re.SubexpIndex("versionNumber")
+			versionNumberAndDate := hashVersionAndDate[idxVersionNumberAndDate]
+			versionNumber := hashVersionAndDate[idxVersionNumber]
+			versionTitles = append(versionTitles, versionNumberAndDate)
+			versionNumbers = append(versionNumbers, versionNumber)
+			versionLineNumbers = append(versionLineNumbers, lineCount)
+		}
+		lineCount++
+	}
+	if scanner.Err() != nil {
+		fmt.Println(scanner.Err())
+	}
+	return changeLogInfo{versionLineNumbers, changeLogText, changeLogLength, versionTitles, versionNumbers}
+}
+
+func createReleaseNotes(changeLogInfo changeLogInfo) []string {
+	re, err := regexp.Compile(`^\[.+\]: .+`)
+	if err != nil {
+		log.Fatal(err)
+	}
+	lastVersionIdx := len(changeLogInfo.versionLineNumbers) - 1
+	contributorsAndErrors := ""
+	for i := changeLogInfo.versionLineNumbers[lastVersionIdx]; i < changeLogInfo.changeLogLength; i++ {
+		if re.MatchString(changeLogInfo.changeLogText[i]) {
+			contributorsAndErrors += changeLogInfo.changeLogText[i] + "\n"
+		}
+	}
+	var releaseNotes []string
+	// exclude Last version (## 0.2.0) with - 1
+	for i, versionLineNumber := range changeLogInfo.versionLineNumbers[:] {
+		releaseBodyLines := ""
+		if !(i == (lastVersionIdx)) {
+			for j := changeLogInfo.versionLineNumbers[i] + 1; j < changeLogInfo.versionLineNumbers[i+1]; j++ {
+				releaseBodyLines += changeLogInfo.changeLogText[j] + "\n"
+			}
+		} else {
+			// Handle last version (## 0.2.0)
+			if versionLineNumber == changeLogInfo.versionLineNumbers[lastVersionIdx] {
+				for j := 1; j < changeLogInfo.changeLogLength-changeLogInfo.versionLineNumbers[lastVersionIdx]; j++ {
+					currentLineOfText := changeLogInfo.changeLogText[versionLineNumber+j]
+					if !re.MatchString(currentLineOfText) {
+						releaseBodyLines += currentLineOfText + "\n"
+					}
+				}
+			}
+		}
+		releaseNotes = append(releaseNotes, releaseBodyLines+contributorsAndErrors)
+	}
+	return releaseNotes
+}
+
+func makeOrUpdateGitHubRelease(dataForAPI dataForAPI) {
+	// https://docs.github.com/en/rest/reference/releases
+	postBody, err := json.Marshal(map[string]string{
+		"tag_name": dataForAPI.tagForRelease,
+		"name":     dataForAPI.versionTitle,
+		"body":     dataForAPI.releaseNote,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	responseBody := bytes.NewBuffer(postBody)
+	releasesURL := ""
+	if dataForAPI.requestType == "POST" {
+		repoOwner, repoName := splitAndEncodeURLPath(dataForAPI.repoPath)
+		releasesURL = fmt.Sprintf("https://api.github.com/repos/%v/%v/releases", repoOwner, repoName)
+	} else if dataForAPI.requestType == "PATCH" {
+		releasesURL = dataForAPI.urlWithID
+	}
```

Comment:
Nit: This `if` looks weird. It means that `urLWithID` is only used sometimes (in a non-obvious way). I think it'd be less code and clearer code to have the caller build the URL.